packages/components-js/scripts/generateVRTPages.ts

```diff
@@ -177,9 +177,13 @@ const generateVRTPagesForJsFramework = (htmlFileContentMap: Record<string, strin
           ? convertToRemixVRTPage(...baseParams, reactCharacteristics)
           : { fileName: '', fileContent: '' };
 
-      writeFile(path.resolve(pagesDirectories[framework], convertedFileName), convertedFileContent);
-
-      return './' + path.parse(convertedFileName).name;
+      writeFile(
+        framework === 'nextjs'
+          ? path.resolve(path.resolve(rootDirectory, `${pagesDirectories[framework]}/${fileName}`), convertedFileName)
+          : path.resolve(pagesDirectories[framework], convertedFileName),
+        convertedFileContent
+      );
+      return framework === 'nextjs' ? './' + fileName : './' + path.parse(convertedFileName).name;
```

Comment:
Can we reduce complexity and conditions to how it was before? Wouldn't `convertToNextJsVRTPage()` be the right place to do this since it is returning `fileName` as part of its object that is destructured above?
Currently it feels like a step back or around the logical flow.